File: src/kyu_6/time_math/js/timeMath.js
function timeMath(time1, op, time2) {
    let time1Parser = timeParser(time1);
    let time2Parser = timeParser(time2);
    let seconds1 = time1Parser.reduce((a, b) => a + b);
    let seconds2 = time2Parser.reduce((a, b) => a + b);
    let operation = calculate(seconds1, seconds2, op);
    return result(operation);
}

function timeParser(time) {
    let result = [];
    let prepared = time.split(":");
    result.push(prepared[0] * 3600)
    result.push(prepared[1] * 60)
    result.push(prepared[2] * 1)
    return result;
}

function calculate(time1, time2, op) {
    return op === "+" ? ~~time1 + ~~time2 : ~~time1 - ~~time2;
}

function result(operation) {
    let total = operation < 0 ? (23 * 3600 + 59 * 60 + 60) - Math.abs(operation) : operation;
    return formatTime(total / 3600 === 24 ? 0
            : total / 3600 > 24 ? (total / 3600) - 24
                : total / 3600) +
        ":" +
        formatTime((total % 3600) / 60) +
        ":" +
        formatTime(total % 60);
}

function formatTime(time) {
    return Math.trunc(time).toString().length < 2 ? "0" + Math.trunc(time) : Math.trunc(time);
}

module.exports = timeMath;

File: src/kyu_6/time_math/c/time_math.c
#include <stdio.h>

char *time_math(char *time3, const char *time1, char op, const char *time2)
{
    int h1, h2, m1, m2, s1, s2, t3;
    sscanf(time1, "%d:%d:%d", &h1, &m1, &s1);
    sscanf(time2, "%d:%d:%d", &h2, &m2, &s2);
    t3 = (3600 * h1 + 60 * m1 + s1) +
         ('+' == op ? 1 : -1) *
         (3600 * h2 + 60 * m2 + s2);
    if ( t3 < 0 ) t3 += 86400;
    sprintf(time3, "%02d:%02d:%02d", t3 / 3600 % 24, t3 % 3600 / 60 % 60, t3 % 60);
    return time3;
}